tests/test_reader.py

```diff
@@ -0,0 +1,135 @@
+import pytest
+from PIL import Image
+
+from readlif.reader import LifFile
+
+
+def test_lif_file_with_file_path(valid_lif_filepath):
+    lif_image = LifFile(valid_lif_filepath).get_image(0)
+    lif_image_frame = lif_image.get_frame(z=0, t=0, c=0)
+    assert lif_image_frame
+
+
+def test_lif_file_with_file_buffer(valid_lif_filepath):
+    with open(valid_lif_filepath, "rb") as file:
+        lif_image = LifFile(file).get_image(0)
+        lif_image_frame = lif_image.get_frame(z=0, t=0, c=0)
+        assert lif_image_frame
+
+
+def test_lif_file_with_non_lif_files(tmp_path, artifacts_dirpath):
+    with open(tmp_path / "not-a-lif-file.txt", "w") as file:
+        with pytest.raises(ValueError):
+            LifFile(file)
+
+    with pytest.raises(ValueError):
+        LifFile(artifacts_dirpath / "misc" / "valid-tiff.tif")
+
+
+def test_lif_file_with_single_image_lif(valid_single_image_lif_filepath):
+    lif_file = LifFile(valid_single_image_lif_filepath)
+    assert repr(lif_file) == "'LifFile object with 1 image'"
+    assert len(lif_file.image_list) == 1
+    with pytest.raises(ValueError):
+        lif_file.get_image(1)
+
+
+def test_lif_file_get_iter_image(xyzt_example_lif_filepath):
+    images = [image for image in LifFile(xyzt_example_lif_filepath).get_iter_image()]
+    assert len(images) == 1
+
+
+def test_lif_file_with_new_lasx(artifacts_dirpath):
+    """
+    TODO(KC): figure out what is special or "new" about the "new_lasx.lif" file.
+    """
+    lif_file = LifFile(artifacts_dirpath / "misc" / "new_lasx.lif")
+    assert len(lif_file.image_list) == 1
+
+
+def test_lif_image_get_frame_out_of_range_args(xyzt_example_lif_filepath):
+    lif_image = LifFile(xyzt_example_lif_filepath).get_image(0)
+    args_for_get_frame = [(10, 0, 0), (0, 10, 0), (0, 0, 10), (0, 0, 0, 10)]
+    for args in args_for_get_frame:
+        with pytest.raises(ValueError):
+            lif_image.get_frame(*args)
```

Comment:
It's a drop in the bucket but I think this is more readable and diagnosable.

```suggestion
def test_lif_image_get_frame_out_of_range_args(xyzt_example_lif_filepath):
    lif_image = LifFile(xyzt_example_lif_filepath).get_image(0)

    # In each case, overshooting the index raises a ValueError:

    max_z_idx = lif_image.nz - 1
    lif_image.get_frame(z=max_z_idx)
    with pytest.raises(ValueError):
        lif_image.get_frame(z=max_z_idx + 1)

    max_t_idx = lif_image.nt - 1
    lif_image.get_frame(t=max_t_idx)
    with pytest.raises(ValueError):
        lif_image.get_frame(t=max_t_idx + 1)

    max_c_idx = lif_image.channels - 1
    lif_image.get_frame(c=max_c_idx)
    with pytest.raises(ValueError):
        lif_image.get_frame(c=max_c_idx + 1)

    max_m_idx = lif_image.n_mosaic - 1
    lif_image.get_frame(m=max_m_idx)
    with pytest.raises(ValueError):
        lif_image.get_frame(m=max_m_idx + 1)

    # FIXME It is unclear why 100 is enough to be considered out of range.
    with pytest.raises(ValueError):
        lif_image._get_item(100)
```

Reply:
agreed this is more sensible and maintainable. rewrote slightly to
```python
    for (dimension_kwarg, lif_image_attr) in [
        ("z", "nz"),
        ("t", "nt"),
        ("c", "channels"),
        ("m", "n_mosaic"),
    ]:
        max_index_for_dimension = getattr(lif_image, lif_image_attr) - 1
        lif_image.get_frame(**{dimension_kwarg: max_index_for_dimension})
        with pytest.raises(ValueError):
            lif_image.get_frame(**{dimension_kwarg: max_index_for_dimension + 1})
```